Tests/BSGConfigurationBuilderTests.m

```diff
@@ -0,0 +1,494 @@
+#import "BSGConfigurationBuilder.h"
+#import "BugsnagConfiguration.h"
+#import <XCTest/XCTest.h>
+
+@interface BSGConfigurationBuilderTests : XCTestCase
+@end
+
+@implementation BSGConfigurationBuilderTests
+
+- (void)testDecodeEmptyOptions {
+    BugsnagConfiguration *config = [BSGConfigurationBuilder configurationFromOptions:@{}];
+    XCTAssertNil(config);
+}
+
+- (void)testDecodeApiKey {
+    BugsnagConfiguration *config = [BSGConfigurationBuilder
+        configurationFromOptions:@{@"apiKey" : @"foo123"}];
+    XCTAssertEqualObjects(@"foo123", config.apiKey);
+}
+
+- (void)testDecodeDefaultAutoNotify {
```

Comment:
These tests could be probably be condensed into one method as they're all testing the default values - or alternatively, the check for an API key could be removed from all but one of the `testDecodeDefault*` methods.

Reply:
Keeping the individual tests as narrow as possible makes it easy to see which part of the implementation is failing. In lieu of a complete suite of every possible permutation, these cover the critical paths of setting 1-2 fields (including API key) and ensuring correct behavior.

Smaller tests can also produce better results when enabling test execution randomization.